qdrant_client/local/payload_filters.py

```diff
@@ -86,18 +87,40 @@ def check_geo_polygon(condition: models.GeoPolygon, values: Any) -> bool:
     return False
 
 
+def check_range_interface(condition: models.RangeInterface, value: Any) -> bool:
+    if isinstance(condition, models.Range):
+        return check_range(condition, value)
+    if isinstance(condition, models.DatetimeRange):
+        return check_datetime_range(condition, value)
+    return False
+
+
 def check_range(condition: models.Range, value: Any) -> bool:
     if not isinstance(value, (int, float)):
         return False
-    if condition.lt is not None and value >= condition.lt:
-        return False
-    if condition.lte is not None and value > condition.lte:
-        return False
-    if condition.gt is not None and value <= condition.gt:
+    return (
+        (condition.lt is None or value < condition.lt)
+        and (condition.lte is None or value <= condition.lte)
+        and (condition.gt is None or value > condition.gt)
+        and (condition.gte is None or value >= condition.gte)
+    )
+
+
+def check_datetime_range(condition: models.DatetimeRange, value: Any) -> bool:
+    if not isinstance(value, str):
```

Comment:
don't we accept datetime objects?

Reply:
can we have datetime objects in the payload?

Reply:
users can add datetime objects to payload, however pydantic will convert them to strings

so we need to store strings in payload, but interface should support working with datetime objects

I think users would expect passing datetime objects to the methods rather than strings

Reply:
so `check_datetime_range` is correct, it does not need to handle datetime

Reply:
I will update the payload fixture then, to account for passing datetime objects to the payload